addon/shield.js

```diff
@@ -0,0 +1,210 @@
+/* globals require, exports, Locale, Task, ClientID */
+"use strict";
+const {Cu} = require("chrome");
+const {PlacesProvider} = require("addon/PlacesProvider");
+const {MetadataStore, METASTORE_NAME} = require("addon/MetadataStore");
+const {TelemetrySender} = require("addon/TelemetrySender");
+const {TabTracker} = require("addon/TabTracker");
+const {ActivityStreams} = require("addon/ActivityStreams");
+const {setTimeout, clearTimeout} = require("sdk/timers");
+const prefs = require("sdk/preferences/service");
+const {getAddonByID} = require("sdk/addon/manager");
+const {PageMod} = require("sdk/page-mod");
+
+Cu.import("resource://gre/modules/ClientID.jsm");
+Cu.import("resource://gre/modules/Task.jsm");
+Cu.import("resource://gre/modules/Locale.jsm");
+
+const {OS} = Cu.import("resource://gre/modules/osfile.jsm", {});
+
+// The constant to set the limit of MetadataStore reconnection
+// The addon will try reconnecting to the database in the next minute periodically,
+// if it fails to establish the connection in the addon initialization
+const kMaxConnectRetry = 120;
+
+function Feature() {}
+
+Feature.prototype = {
+  _options: {},
+  _variant: null,
+  _dlp: null,
+  _tilesPageMod: null,
+  _activityStreamApp: null,
+  _metadataStore: null,
+  _tabTracker: null,
+  _telemetrySender: null,
+  _connectRetried: 0,
+  _reconnectTimeoutID: null,
+  _initializedTiles: false,
+  _doesHaveTestPilot: false,
+  _eligibilityFilters: [],
+  _placesQueries: null,
+  _originalPreloadPref: null,
+
+  setVariant(variant) {
+    this._variant = `shield-study-01-${variant}`;
+  },
+
+  setUp: Task.async(function*() {
+    this._originalPreloadPref = prefs.get("browser.newtab.preload");
+    prefs.set("browser.newtab.preload", false);
+    PlacesProvider.links.init();
+    const clientID = yield ClientID.getClientID();
+    this._options.clientID = clientID;
+    this._options.shield_variant = this._variant;
+    this._options.telemetry = false;
+    this._options.tp_version = "1.1.7";
```

Comment:
i *think* this should reflect the version of the addon that is in test pilot right now, so i guess 1.1.12 ?
@emtwo can you remember why we had this and what this was again?